### PR TITLE
Revamp build process, project structure and workflows 

### PR DESCRIPTION
**Pull request by RDPerera**
## Purpose
This PR will fixes:https://github.com/ballerina-platform/ballerina-library/issues/5973  with ,

1. Integrate tests into the Build Process.
2. Rename dir names to align with other Ballerina Tools.
3. Remove redundant PR template.
4. Implement workflows to PR build.
5. Implement workflows to build with timestamped.
6. Implement workflows to stale PR check.
7. Update the current Release workflow with revamped build process.

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: .github/workflows/build-timestamped-master.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+          - name: Grant execute permission for gradlew
+            run: chmod +x gradlew     
+
+          - name: Build the Package
+            env:
+              packageUser: ${{ github.actor }}
+              packagePAT: ${{ secrets.GITHUB_TOKEN }}
+            run: |
+                ./gradlew fullbuild
```
Comment (niveathika) [inline .github/workflows/build-timestamped-master.yml:54]: Shall we use standard task names? `./gradlew build`

File: build.gradle [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```
Comment (niveathika) [inline build.gradle:2]: No need to change the license year for edited files.

File: .github/workflows/build-timestamped-master.yml (hunk 1 above)
Comment (RDPerera) [reply .github/workflows/build-timestamped-master.yml:54]: If a developer has previously installed EDI tool, the `./gradlew build` command already available for building the package. This means the developer should already have the `bal-tools.toml `file in their `.ballerina/.config/` directory.

On the other hand, `./gradlew fullbuild` is intended for running PR and Release workflows that create a clean `bal-tools.toml` in the `.ballerina/.config/` directory on GitHub's VM.

The reason to having a separate task for that is, if we create a clean `bal-tools.toml`(with only allowing EDI Tool) in developer's `.ballerina/.config/` directory, the existing tools that he used other than EDI tool will be unusable.

File: build.gradle (hunk 2 above)
Comment (RDPerera) [reply build.gradle]: Sure, Changed with [b20f978](https://github.com/ballerina-platform/edi-tools/pull/28/commits/b20f97803a9c3b57c7e0e33ad3d4cc088062593c).

**Review by niveathika — APPROVED** [review]

**Pull request merged**